Library/Homebrew/test/build_environment_spec.rb

```diff
@@ -0,0 +1,70 @@
+require "build_environment"
+
+RSpec::Matchers.define :use_userpaths do
+  match do |actual|
+    actual.userpaths?
+  end
+end
+
+describe BuildEnvironment do
+  let(:env) { described_class.new }
+
+  describe "#<<" do
+    it "returns itself" do
+      expect(env << :foo).to eq env
+    end
+  end
+
+  describe "#merge" do
+    it "returns itself" do
+      expect(env.merge([])).to eq env
+    end
+  end
+
+  describe "#std?" do
+    it "returns true if the environment contains :std" do
+      expect(env << :std).to be_std
+    end
+
+    it "returns false if the environment does not contain :std" do
+      expect(env).not_to be_std
+    end
+  end
+
+  describe "#userpaths?" do
+    it "returns true if the environment contains :userpaths" do
+      expect(env << :userpaths).to use_userpaths
+    end
+
+    it "returns false if the environment does not contain :userpaths" do
+      expect(env).not_to use_userpaths
+    end
+  end
+end
+
+describe BuildEnvironmentDSL do
+  subject {
+    Object.new.extend(BuildEnvironmentDSL).tap { |o| o.instance_eval(&block) }
+  }
+
+  context "single argument" do
+    let(:block) {
+      proc do
+        env :userpaths
+      end
+    }
```

Comment:
I don't like defining methods inside specs, that is specifically what procs and lambda are for. Anyway, in this case it really only works with procs, not with lambdas or methods.